--- Contents/Resources/share/themes/Adwaita/gtk-2.0/main.rc.orig	2017-02-07 13:40:44.000000000 +0100
+++ Contents/Resources/share/themes/Adwaita/gtk-2.0/main.rc	2017-02-11 19:33:53.000000000 +0100
@@ -2414,6 +2432,15 @@
 
 }
 
+style "toolbar_button_unclipped" {
+
+  xthickness = 4
+  ythickness = 4
+
+  GtkButton::inner-border = {0, 0, 0, 0}
+
+}
+
 style "toolbar_separator" {
 
   GtkWidget::wide-separators  = 1
@@ -2536,7 +2564,7 @@
 # have an entry inside
 
 # Toolbar buttons have different paddings
-widget_class "*<GtkToolbar>*<GtkButton>"                           style "toolbar_button"
+widget_class "*<GtkToolbar>*<GtkButton>"                           style "toolbar_button_unclipped"
 
 # Notebooks are white, act accordingly
 widget_class "*<GtkNotebook>*<GtkEntry>"                           style "notebook_entry"
