diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/dialogs/dialog-events.cpp ./src/dialogs/dialog-events.cpp
--- ..\inkscape_46_patch-1/src/dialogs/dialog-events.cpp	2008-03-28 11:59:30.859375000 +0100
+++ ./src/dialogs/dialog-events.cpp	2008-03-28 14:17:18.328125000 +0100
@@ -156,13 +156,10 @@
 #endif
 
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2 );
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else
-        transient_policy = 0;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif
 
     if (transient_policy) {
 
@@ -185,13 +182,10 @@
                            SPDesktop *desktop, win_data *wd )
 {
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2);
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else
-        return;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif
 
     if (!transient_policy)
         return;
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/dock-behavior.cpp ./src/ui/dialog/dock-behavior.cpp
--- ..\inkscape_46_patch-1/src/ui/dialog/dock-behavior.cpp	2008-03-28 12:00:13.484375000 +0100
+++ ./src/ui/dialog/dock-behavior.cpp	2008-03-28 14:17:18.328125000 +0100
@@ -215,13 +215,10 @@
 DockBehavior::onDesktopActivated(SPDesktop *desktop)
 {
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2);
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else    
-        return;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif        
 
     if (!transient_policy) 
         return;
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/floating-behavior.cpp ./src/ui/dialog/floating-behavior.cpp
--- ..\inkscape_46_patch-1/src/ui/dialog/floating-behavior.cpp	2008-03-28 12:00:13.656250000 +0100
+++ ./src/ui/dialog/floating-behavior.cpp	2008-03-28 14:17:18.328125000 +0100
@@ -96,13 +96,10 @@
 FloatingBehavior::onDesktopActivated (SPDesktop *desktop)
 {
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2);
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else    
-        return;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif        
 
     if (!transient_policy) 
         return;
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.cpp ./src/ui/dialog/inkscape-preferences.cpp
--- ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.cpp	2008-03-28 12:00:13.000000000 +0100
+++ ./src/ui/dialog/inkscape-preferences.cpp	2008-03-28 16:34:47.531250000 +0100
@@ -469,13 +469,8 @@
     _win_ontop_none.init ( _("None"), "options.transientpolicy", "value", 0, false, 0);
     _win_ontop_normal.init ( _("Normal"), "options.transientpolicy", "value", 1, true, &_win_ontop_none);
     _win_ontop_agressive.init ( _("Aggressive"), "options.transientpolicy", "value", 2, false, &_win_ontop_none);
 
-// FIXME: Temporary Win32 special code to enable transient dialogs
-#ifdef WIN32
-    _win_ontop_win32.init ( _("Dialogs stay on top (experimental!)"), "options.dialogsontopwin32", "value", false);
-#endif
-
     _page_windows.add_group_header( _("Saving window geometry (size and position):"));
     _page_windows.add_line( false, "", _win_save_geom_off, "",
                             _("Let the window manager determine placement of all windows"));
     _page_windows.add_line( false, "", _win_save_geom_prefs, "",
@@ -488,19 +483,17 @@
                             _("Dockable"));
     _page_windows.add_line( true, "", _win_floating, "",
                             _("Floating"));
 
-    _page_windows.add_group_header( _("Dialogs on top:"));
-#ifndef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
+#ifndef WIN32 // non-Win32 special code to enable transient dialogs
+	_page_windows.add_group_header( _("Dialogs on top:"));
+
     _page_windows.add_line( true, "", _win_ontop_none, "",
                             _("Dialogs are treated as regular windows"));
     _page_windows.add_line( true, "", _win_ontop_normal, "",
                             _("Dialogs stay on top of document windows"));
     _page_windows.add_line( true, "", _win_ontop_agressive, "",
                             _("Same as Normal but may work better with some window managers"));
-#else
-    _page_windows.add_line( false, "", _win_ontop_win32, "",
-                            _("Whether dialogs should stay on top of document windows. Read the ReleaseNotes on this issue! (Rightclick the taskbar button and press 'Restore' to bring back a minimized document window)"));
 #endif
 
     _page_windows.add_group_header( _("Miscellaneous:"));
 #ifndef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.h ./src/ui/dialog/inkscape-preferences.h
--- ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.h	2008-03-28 12:00:13.671875000 +0100
+++ ./src/ui/dialog/inkscape-preferences.h	2008-03-28 14:17:18.343750000 +0100
@@ -131,13 +131,8 @@
     PrefRadioButton _win_ontop_none, _win_ontop_normal, _win_ontop_agressive;
     PrefRadioButton _win_save_geom_off, _win_save_geom, _win_save_geom_prefs;
     PrefCheckButton _win_hide_task, _win_zoom_resize , _win_show_close;
 
-// FIXME: Temporary Win32 special code to enable transient dialogs
-#ifdef WIN32 
-    PrefCheckButton _win_ontop_win32;   
-#endif    
-
     PrefCheckButton _calligrapy_use_abs_size;
     PrefCheckButton _calligrapy_keep_selected;
 
     PrefCheckButton _connector_ignore_text;
