Only in .: 04.patch
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/dialogs/dialog-events.cpp ./src/dialogs/dialog-events.cpp
--- ..\inkscape_46_patch-1/src/dialogs/dialog-events.cpp	2008-03-27 16:03:57.953125000 +0100
+++ ./src/dialogs/dialog-events.cpp	2008-03-27 16:13:30.593750000 +0100
@@ -156,13 +156,10 @@
 #endif
 
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2 );
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else
-        transient_policy = 0;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif
 
     if (transient_policy) {
 
@@ -185,13 +182,10 @@
                            SPDesktop *desktop, win_data *wd )
 {
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2);
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else
-        return;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif
 
     if (!transient_policy)
         return;
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/dock-behavior.cpp ./src/ui/dialog/dock-behavior.cpp
--- ..\inkscape_46_patch-1/src/ui/dialog/dock-behavior.cpp	2008-03-27 16:04:56.906250000 +0100
+++ ./src/ui/dialog/dock-behavior.cpp	2008-03-27 16:14:34.421875000 +0100
@@ -215,13 +215,10 @@
 DockBehavior::onDesktopActivated(SPDesktop *desktop)
 {
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2);
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else    
-        return;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif        
 
     if (!transient_policy) 
         return;
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/floating-behavior.cpp ./src/ui/dialog/floating-behavior.cpp
--- ..\inkscape_46_patch-1/src/ui/dialog/floating-behavior.cpp	2008-03-27 16:04:57.093750000 +0100
+++ ./src/ui/dialog/floating-behavior.cpp	2008-03-27 16:15:15.703125000 +0100
@@ -96,13 +96,10 @@
 FloatingBehavior::onDesktopActivated (SPDesktop *desktop)
 {
     gint transient_policy = prefs_get_int_attribute_limited ( "options.transientpolicy", "value", 1, 0, 2);
 
-#ifdef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
-    if (prefs_get_int_attribute ( "options.dialogsontopwin32", "value", 0))
-        transient_policy = 2;
-    else    
-        return;
+#ifdef WIN32 // Win32 special code to enable transient dialogs
+    transient_policy = 2;
 #endif        
 
     if (!transient_policy) 
         return;
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.cpp ./src/ui/dialog/inkscape-preferences.cpp
--- ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.cpp	2008-03-27 16:04:56.687500000 +0100
+++ ./src/ui/dialog/inkscape-preferences.cpp	2008-03-27 16:19:14.843750000 +0100
@@ -488,19 +488,17 @@
                             _("Dockable"));
     _page_windows.add_line( true, "", _win_floating, "",
                             _("Floating"));
 
-    _page_windows.add_group_header( _("Dialogs on top:"));
-#ifndef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
+#ifndef WIN32 // non-Win32 special code to enable transient dialogs
+	_page_windows.add_group_header( _("Dialogs on top:"));
+
     _page_windows.add_line( true, "", _win_ontop_none, "",
                             _("Dialogs are treated as regular windows"));
     _page_windows.add_line( true, "", _win_ontop_normal, "",
                             _("Dialogs stay on top of document windows"));
     _page_windows.add_line( true, "", _win_ontop_agressive, "",
                             _("Same as Normal but may work better with some window managers"));
-#else
-    _page_windows.add_line( false, "", _win_ontop_win32, "",
-                            _("Whether dialogs should stay on top of document windows. Read the ReleaseNotes on this issue! (Rightclick the taskbar button and press 'Restore' to bring back a minimized document window)"));
 #endif
 
     _page_windows.add_group_header( _("Miscellaneous:"));
 #ifndef WIN32 // FIXME: Temporary Win32 special code to enable transient dialogs
diff -r -u4 -x .svn ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.h ./src/ui/dialog/inkscape-preferences.h
--- ..\inkscape_46_patch-1/src/ui/dialog/inkscape-preferences.h	2008-03-27 16:04:57.125000000 +0100
+++ ./src/ui/dialog/inkscape-preferences.h	2008-03-27 16:19:42.250000000 +0100
@@ -131,13 +131,8 @@
     PrefRadioButton _win_ontop_none, _win_ontop_normal, _win_ontop_agressive;
     PrefRadioButton _win_save_geom_off, _win_save_geom, _win_save_geom_prefs;
     PrefCheckButton _win_hide_task, _win_zoom_resize , _win_show_close;
 
-// FIXME: Temporary Win32 special code to enable transient dialogs
-#ifdef WIN32 
-    PrefCheckButton _win_ontop_win32;   
-#endif    
-
     PrefCheckButton _calligrapy_use_abs_size;
     PrefCheckButton _calligrapy_keep_selected;
 
     PrefCheckButton _connector_ignore_text;
