Index: share/extensions/perspective.py
===================================================================
--- share/extensions/perspective.py	(revision 18378)
+++ share/extensions/perspective.py	(working copy)
@@ -68,7 +68,7 @@
             file = self.args[-1]
             id = self.options.ids[0]
             for query in q.keys():
-                _,f,err = os.popen3("inkscape --query-%s --query-id=%s %s" % (query,id,file))
+                _,f,err = os.popen3('inkscape --query-%s --query-id=%s "%s"' % (query,id,file))
                 q[query] = float(f.read())
                 f.close()
                 err.close()
Index: share/extensions/summersnight.py
===================================================================
--- share/extensions/summersnight.py	(revision 18378)
+++ share/extensions/summersnight.py	(working copy)
@@ -47,7 +47,7 @@
             file = self.args[-1]
             id = self.options.ids[0]
             for query in self.q.keys():
-                _,f,err = os.popen3("inkscape --query-%s --query-id=%s %s" % (query,id,file))
+                _,f,err = os.popen3('inkscape --query-%s --query-id=%s "%s"' % (query,id,file))
                 self.q[query] = float(f.read())
                 f.close()
                 err.close()
